src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -2379,6 +2379,111 @@ paths:
                 $ref: '#/components/schemas/ErrorModel'
       x-codegen-request-body-name: upgrade
 
+##############################################################################
+## SEARCH API STANDARD ENDPOINTS
+##############################################################################
+  /api/repository/v1/search/{id}/index:
```

Comment:
It's unfortunate that `index` is both a noun and a verb. `create` is too general. Thinking about this using REST conventions, maybe the noun is `index` not search? E.g. `/index/create` and `/index/query` or `/index/search`. I don't want to hold up the PR on this discussion though; since this isn't yet a public API we can spend some time iterating on naming, return types, etc, in future PRs.